src/setup/cortx_deploy

```diff
@@ -110,26 +113,60 @@ node_id=`cat /etc/machine-id`
 cluster_id=$(conf_get $(yaml_config $CLUSTER_INFO) "cluster>id")
 log $INFO "Starting provisioning on node $node_id which is part of cluster $cluster_id."
 
-# Apply Cluster Config
-log $INFO "cortx_setup config apply -f $(yaml_config $CLUSTER_INFO) -c '$CORTX_CONFSTORE' $FORCE_PROVISIONING"
-cortx_setup config apply -f $(yaml_config $CLUSTER_INFO) -c "$CORTX_CONFSTORE" $FORCE_PROVISIONING;
-
-# Apply Common Config
-log $INFO "cortx_setup config apply -f $(yaml_config $CONFIG_INFO) -c '$CORTX_CONFSTORE' $FORCE_PROVISIONING"
-cortx_setup config apply -f $(yaml_config $CONFIG_INFO) -c "$CORTX_CONFSTORE" $FORCE_PROVISIONING;
-
-# Validate config
-if ! [ -z $VALIDATE_CONF ];
-then
-    log $INFO "cortx_setup config validate -f $(yaml_config $CLUSTER_INFO) -c '$CORTX_CONFSTORE'"
-    cortx_setup config validate -f $(yaml_config $CLUSTER_INFO) -c "$CORTX_CONFSTORE" || exit 1
-    log $INFO "cortx_setup config validate -f $(yaml_config $CONFIG_INFO) -c '$CORTX_CONFSTORE'"
-    cortx_setup config validate -f $(yaml_config $CONFIG_INFO) -c "$CORTX_CONFSTORE" || exit 1
+# Check current image version
+current_image_version=$(conf_get $CORTX_CONFSTORE "cortx>common>release>version")
+# if current version is not available in confstore
+if [ $current_image_version = "[null]" ]; then
+    DEPLOY_FLAG=true
+else
+    # Below can be used when subversioning is considered.
+    #current_primary_ver=$(awk -F. '{print $1 "." $2 "." $3}' <<< $current_image_version)
+    current_version=$(awk -F. '{print $4}' <<< $current_image_version)
 fi
 
-# Bootstrap Cluster
-log $INFO "cortx_setup cluster bootstrap -c '$CORTX_CONFSTORE' $FORCE_PROVISIONING"
-cortx_setup cluster bootstrap -c "$CORTX_CONFSTORE" $FORCE_PROVISIONING;
+update_image_version=$(sed -e 's/[][]//g' -e 's/"//g' <<< $(conf_get $RELEASE_INFO "VERSION"))
+# Below can be used when subversioning is considered.
+#image_primary_ver=$(awk -F- '{print $1}' <<<  $update_image_version )
+update_version=$(awk -F- '{print $2}' <<<  $update_image_version)
+
+if [ $current_version -eq $update_version ]; then
+    if [ ! -z $FORCE_PROVISIONING ]; then
+        DEPLOY_FLAG=true
+    else
+        return 0
+    fi
+elif [ $current_version -gt $update_version ]; then
+    log $ERROR "Cannot update Cortx Image current_version ($current_image_version) > update_version ($update_image_version)"
+    exit 1
+else
+    DEPLOY_FLAG=false
```

Comment:
No need to maintain multiple flags as indicated earlier. There will be 5 more commands added in future and this logic will be more complex. Simplify it... 

ACTION=upgrade | deploy | downgrade | replace | recover | etc.

Reply:
Done.

Reply:
Done.